templates/image-editor/editor.html.twig

```diff
@@ -0,0 +1,221 @@
+            <div id="toggle_preview"></div>
+        </div>
+
+        <div class="colors block">
+            <h2 class="trn toggle" data-target="toggle_colors">Colors</h2>
+            <input
+                    title="Click to change color"
+                    type="color"
+                    class="color_area"
+                    id="main_color"
+                    value="#0000ff"	/>
+            <div class="content" id="toggle_colors"></div>
+        </div>
+
+        <div class="block" id="info_base">
+            <h2 class="trn toggle toggle-full" data-target="toggle_info">Information</h2>
+            <div class="content" id="toggle_info"></div>
+        </div>
+
+        <div class="details block" id="details_base">
+            <h2 class="trn toggle toggle-full" data-target="toggle_details">Layer details</h2>
+            <div class="content" id="toggle_details"></div>
+        </div>
+
+        <div class="layers block">
+            <h2 class="trn">Layers</h2>
+            <div class="content" id="layers_base"></div>
+        </div>
+    </div>
+</div>
+<div class="mobile_menu">
+    <button class="left_mobile_menu" id="left_mobile_menu_button" type="button">
+        <span class="sr_only">Toggle Menu</span>
+    </button>
+    <button class="right_mobile_menu" id="mobile_menu_button" type="button">
+        <span class="sr_only">Toggle Menu</span>
+    </button>
+</div>
+<div class="hidden" id="tmp"></div>
+<div id="popups"></div>
+
+{% set imageFileExtension = pimcore_file_extension(asset.getFilename()) %}
+{% set imageUrl = path('pimcore_studio_api_stream_image', {id: asset.getId()}) %}
+
+{% if imageFileExtension not in ['png', 'jpg', 'jpeg'] %}
+    {% set imageUrl = path('pimcore_studio_api_stream_image_custom', {id: asset.getId(), mimeType: 'png', resizeMode: 'none' }) %}
+{% endif %}
+
+<img style="visibility: hidden" id='image' src="{{ imageUrl }}" alt="editor preview" />
+<script {{ pimcore_csp.getNonceHtmlAttribute()|raw }}>
+
+    /**
+     * wait for image editor to be fully available
+     * before loading image to editor
+     */
+    async function loadEditor(e) {
+        return new Promise(resolve => {
+            var checkInterval = setInterval(async () => {
+                if (window.Layers) {
+                    clearInterval(checkInterval);
+
+                    const translationKeys = await getTranslation()
+                    manipulateMenuBar(translationKeys);
+
+                    loadImageToEditor(e);
+
+                    resolve(true);
+                }
+            }, 300);
+        });
+    }
+
+    async function getTranslation() {
+        const url = '{{ path('pimcore_studio_api_translations') }}';
+
+        const response = await fetch(url, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({ locale: 'en', keys: [] })
```

Comment:
We should add the requested translation keys here in the keys array. With this approach we would only get the relevant translation keys instead of the full list.